src/TLuaInterpreter.cpp

```diff
@@ -12083,6 +12151,75 @@ void TLuaInterpreter::initLuaGlobals()
     //FIXME make function call in destructor lua_close(L);
 }
 
+// initialised a slimmed-down Lua state just to run the indenter in a separate sandbox
+// the indenter by default pollutes the global environment with some utility functions
+// and we don't want to tie ourselves to it by exposing them for scripting
+void TLuaInterpreter::initIndenterGlobals()
+{
+    pIndenterState = newstate();
+    storeHostInLua(pIndenterState, mpHost);
+
+    luaL_openlibs(pIndenterState);
+
+    lua_pushstring(pIndenterState, "SESSION");
+    lua_pushnumber(pIndenterState, mHostID);
+    lua_settable(pIndenterState, LUA_GLOBALSINDEX);
+
+    lua_pushstring(pIndenterState, "SCRIPT_NAME");
+    lua_pushstring(pIndenterState, "Lua Indenter Interpreter");
+    lua_settable(pIndenterState, LUA_GLOBALSINDEX);
+
+    lua_pushstring(pIndenterState, "SCRIPT_ID");
+    lua_pushnumber(pIndenterState, -2); // ID 2 is used to indicate that this is the indenter Lua interpreter
+    lua_settable(pIndenterState, LUA_GLOBALSINDEX);
+    lua_register(pIndenterState, "echo", TLuaInterpreter::Echo);
+    lua_register(pIndenterState, "tempTimer", TLuaInterpreter::tempTimer);
+    lua_register(pIndenterState, "send", TLuaInterpreter::sendRaw);
+    lua_register(pIndenterState, "debugc", TLuaInterpreter::debug);
+    // PLACEMARKER: End of Lua functions registration
```

Comment:
Can you add `(indenter)` somewhere in that comment - to distinguish it from the main one in the Qt Creator `ToDo Entries` window?

How does this handle the other "_unregistered_" Mudlet provided functions when they crop up in the scripts being parsed to indent? Does it need to? I'm asking as I do not have a handle on how/whether they need to be _defined_ to enable the magic to happen, or whether they just have to _look_ like they are functions?

Reply:
Fixed.

Unregistered functions get formatted fine.